language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -307,6 +314,28 @@ public FieldAccessCompletionResolver(PositionedOperationContext context) {
                         .collect(Collectors.toList());
                 visibleEntries.addAll(methodSymbols);
                 break;
+            case UNION:
+                if (node.parent().kind() != SyntaxKind.OPTIONAL_FIELD_ACCESS
+                        || ((UnionTypeSymbol) rawType).memberTypeDescriptors().size() != 2) {
+                    break;
+                }
+                List<TypeSymbol> members = ((UnionTypeSymbol) rawType).memberTypeDescriptors().stream()
+                        .map(CommonUtil::getRawType)
+                        .collect(Collectors.toList());
+                List<TypeDescKind> memberTypes = members.stream()
+                        .map(TypeSymbol::typeKind)
+                        .collect(Collectors.toList());
+                if (!memberTypes.contains(TypeDescKind.NIL) || !memberTypes.contains(TypeDescKind.RECORD)) {
```

Comment:
We can rewrite line 325 to 330 as below since we know the member list size is 2:
```java
if (members.stream().allMatch(typeSymbol -> typeSymbol.typeKind() == TypeDescKind.NIL || typeSymbol.typeKind() == TypeDescKind.RECORD)) {
   break;
}
```

Reply:
Fixed